aten/src/ATen/native/Onehot.cpp

```diff
@@ -0,0 +1,35 @@
+#include <ATen/ATen.h>
+
+namespace at { namespace native {
+
+Tensor one_hot(const Tensor &self, int64_t num_classes) {
+    AT_ASSERTM(self.dtype() == kLong, "one_hot is only applicable to index tensor.");
```

Comment:
AT_ERROR is better, iirc AT_ASSERTM tells the user to submit a bug to pytorch

Reply:
@zou3519 Do you mean `AT_CHECK`? I don't think `AT_ERROR` takes any condition as arg, it simply report there is an error unconditionally.